packages/react-components/src/components/utils/responsive.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { RefObject, useEffect, useState } from 'react';
+
+/**
+ * A utility hook for providing the width of a parent element.
+ * Returns updated width if parent/window resizes.
+ * @param containerRef - Ref of a parent element whose width will be returned.
+ */
+export const useContainerWidth = (containerRef: RefObject<HTMLDivElement>): number => {
+  const [width, setWidth] = useState(0);
+  useEffect(() => {
+    const updateWidth = (): void => {
+      const width = containerRef.current?.offsetWidth ?? 0;
```

Comment:
nit: I think use `clientWidth/Height` instead of offsetWidth/Height (as we don't care if borders change size, and we want to capture size changes if a scrollbar changes the interior sizing)
![image](https://user-images.githubusercontent.com/2684369/135540842-6987c7d0-c1c8-453c-82a2-d343cbd17909.png)

Reply:
For this, we need the attribute that gives us the content area including everything. OR we might run into a bug.
Kinda hard to explain in the chat. Can show you in a call if you are interested.